src/Execution.cpp

```diff
@@ -2297,7 +2297,7 @@ GenericValue Interpreter::getConstantExprValue (ConstantExpr *CE,
                                gep_type_end(CE), SF);
   case Instruction::FCmp:
   case Instruction::ICmp:
-    return executeCmpInst(CE->getPredicate(),
+    return executeCmpInst(llvm::dyn_cast<llvm::CmpInst>(CE)->getPredicate(),
```

Comment:
You should use `llvm::cast` when the cast must succeed. It [contains](https://llvm.org/docs/ProgrammersManual.html#the-isa-cast-and-dyn-cast-templates) an assert that the cast is legal, rather than returning null if it isn't. Ditto below